torch/csrc/autograd/autograd.cpp

```diff
@@ -93,6 +93,12 @@ variable_list run_backward(
       if (!grad_fn) {
         grad_fn = impl::try_get_grad_accumulator(input);
       }
+      if (accumulate_grad) {
+        TORCH_CHECK(
```

Comment:
Why do you need to add this test? It's not already covered in the previous PR?

Reply:
We did have this check at one point in the previous PR, but we ended up removing it because it wasn't necessary for python api changes. So this check should be necessary for the cpp api.

Reply:
Ho sounds good!